libkbfs/block_retrieval_queue.go

```diff
@@ -74,18 +78,38 @@ func newBlockRetrievalQueue(numWorkers int) *blockRetrievalQueue {
 // notifyWorker notifies workers that there is a new request for processing.
 func (brq *blockRetrievalQueue) notifyWorker() {
 	go func() {
-		// Get the next queued worker
-		ch := <-brq.workerQueue
-		// Prevent interference with the heap while we're retrieving from it
-		brq.mtx.Lock()
-		defer brq.mtx.Unlock()
-		// Pop from the heap
-		ch <- heap.Pop(brq.heap).(*blockRetrieval)
+		select {
+		case <-brq.doneCh:
+			// Prevent interference with the heap while we're retrieving from it
+			brq.mtx.Lock()
+			defer brq.mtx.Unlock()
+			if brq.heap.Len() > 0 {
```

Comment:
I originally intended that each goroutine will close one request, but you're probably right. Might as well have one do it.